src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -2542,8 +2542,8 @@ exprt c_typecheck_baset::do_special_functions(
       throw 0;
     }
 
-    equal_exprt equality_expr;
-    equality_expr.operands()=expr.arguments();
+    equal_exprt equality_expr(
+      expr.arguments().front(), expr.arguments().back());
```

Comment:
Are we sure this has two elements?

Reply:
That's check 7 lines earlier so this should be ok.